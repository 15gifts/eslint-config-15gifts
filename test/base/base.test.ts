import { getResultJson } from "../helpers";

describe("Base config", () => {
  test("That the base 'valid' file passes", async () => {
    const [result] = await getResultJson(["test/base/valid.js"]);

    expect(result.messages).toMatchSnapshot();
    expect(result.errorCount).toBe(0);
  });

  test("That the base 'invalid' file throws lint errors", async () => {
    const [result] = await getResultJson(["test/base/invalid.js"]);

    expect(result.messages).toMatchSnapshot();
    expect(result.errorCount).toBe(13);
  });
});
