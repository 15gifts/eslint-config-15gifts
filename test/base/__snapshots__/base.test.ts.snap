// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base config That the base 'invalid' file throws lint errors 1`] = `
[
  {
    "column": 15,
    "endColumn": 16,
    "endLine": 5,
    "fix": {
      "range": [
        117,
        134,
      ],
      "text": "defaults
  const",
    },
    "line": 5,
    "message": "Extra semicolon.",
    "messageId": "extraSemi",
    "nodeType": "VariableDeclaration",
    "ruleId": "@typescript-eslint/semi",
    "severity": 2,
  },
  {
    "column": 28,
    "endColumn": 34,
    "endLine": 6,
    "fix": {
      "range": [
        154,
        160,
      ],
      "text": "'golf'",
    },
    "line": 6,
    "message": "Strings must use singlequote.",
    "messageId": "wrongQuotes",
    "nodeType": "Literal",
    "ruleId": "quotes",
    "severity": 2,
  },
  {
    "column": 36,
    "endColumn": 44,
    "endLine": 6,
    "fix": {
      "range": [
        162,
        170,
      ],
      "text": "'tennis'",
    },
    "line": 6,
    "message": "Strings must use singlequote.",
    "messageId": "wrongQuotes",
    "nodeType": "Literal",
    "ruleId": "quotes",
    "severity": 2,
  },
  {
    "column": 45,
    "endColumn": 46,
    "endLine": 6,
    "fix": {
      "range": [
        170,
        180,
      ],
      "text": "]
  const",
    },
    "line": 6,
    "message": "Extra semicolon.",
    "messageId": "extraSemi",
    "nodeType": "VariableDeclaration",
    "ruleId": "@typescript-eslint/semi",
    "severity": 2,
  },
  {
    "column": 27,
    "endColumn": 35,
    "endLine": 7,
    "fix": {
      "range": [
        199,
        207,
      ],
      "text": "'reggae'",
    },
    "line": 7,
    "message": "Strings must use singlequote.",
    "messageId": "wrongQuotes",
    "nodeType": "Literal",
    "ruleId": "quotes",
    "severity": 2,
  },
  {
    "column": 37,
    "endColumn": 44,
    "endLine": 7,
    "fix": {
      "range": [
        209,
        216,
      ],
      "text": "'blues'",
    },
    "line": 7,
    "message": "Strings must use singlequote.",
    "messageId": "wrongQuotes",
    "nodeType": "Literal",
    "ruleId": "quotes",
    "severity": 2,
  },
  {
    "column": 46,
    "endColumn": 59,
    "endLine": 7,
    "fix": {
      "range": [
        218,
        231,
      ],
      "text": "'death metal'",
    },
    "line": 7,
    "message": "Strings must use singlequote.",
    "messageId": "wrongQuotes",
    "nodeType": "Literal",
    "ruleId": "quotes",
    "severity": 2,
  },
  {
    "column": 60,
    "endColumn": 61,
    "endLine": 7,
    "fix": {
      "range": [
        231,
        243,
      ],
      "text": "]

  return",
    },
    "line": 7,
    "message": "Extra semicolon.",
    "messageId": "extraSemi",
    "nodeType": "VariableDeclaration",
    "ruleId": "@typescript-eslint/semi",
    "severity": 2,
  },
  {
    "column": 5,
    "endColumn": 9,
    "endLine": 11,
    "fix": {
      "range": [
        250,
        342,
      ],
      "text": "hair: \`\${hairLength} and \${hairColor}\`,
    hobbies: [...favouriteSports, ...favouriteMusic]",
    },
    "line": 11,
    "message": "Expected object keys to be in natural ascending order. 'hair' should be before 'hobbies'.",
    "nodeType": "Property",
    "ruleId": "sort-keys-fix/sort-keys-fix",
    "severity": 2,
  },
  {
    "column": 5,
    "endColumn": 8,
    "endLine": 12,
    "fix": {
      "range": [
        304,
        366,
      ],
      "text": "age: parseInt(age),
    hair: \`\${hairLength} and \${hairColor}\`",
    },
    "line": 12,
    "message": "Expected object keys to be in natural ascending order. 'age' should be before 'hair'.",
    "nodeType": "Property",
    "ruleId": "sort-keys-fix/sort-keys-fix",
    "severity": 2,
  },
  {
    "column": 10,
    "endColumn": 23,
    "endLine": 12,
    "line": 12,
    "message": "Missing radix parameter.",
    "messageId": "missingRadix",
    "nodeType": "CallExpression",
    "ruleId": "radix",
    "severity": 2,
    "suggestions": [
      {
        "desc": "Add radix parameter \`10\` for parsing decimal numbers.",
        "fix": {
          "range": [
            365,
            365,
          ],
          "text": ", 10",
        },
        "messageId": "addRadixParameter10",
      },
    ],
  },
  {
    "column": 4,
    "endColumn": 5,
    "endLine": 14,
    "fix": {
      "range": [
        380,
        384,
      ],
      "text": "}
}",
    },
    "line": 14,
    "message": "Extra semicolon.",
    "messageId": "extraSemi",
    "nodeType": "ReturnStatement",
    "ruleId": "@typescript-eslint/semi",
    "severity": 2,
  },
  {
    "column": 2,
    "endColumn": 3,
    "endLine": 15,
    "fix": {
      "range": [
        383,
        392,
      ],
      "text": "}

const",
    },
    "line": 15,
    "message": "Extra semicolon.",
    "messageId": "extraSemi",
    "nodeType": "VariableDeclaration",
    "ruleId": "@typescript-eslint/semi",
    "severity": 2,
  },
  {
    "column": 25,
    "endColumn": 32,
    "endLine": 17,
    "fix": {
      "range": [
        411,
        418,
      ],
      "text": "'barry'",
    },
    "line": 17,
    "message": "Strings must use singlequote.",
    "messageId": "wrongQuotes",
    "nodeType": "Literal",
    "ruleId": "quotes",
    "severity": 2,
  },
  {
    "column": 19,
    "endColumn": 25,
    "endLine": 18,
    "fix": {
      "range": [
        440,
        446,
      ],
      "text": "'long'",
    },
    "line": 18,
    "message": "Strings must use singlequote.",
    "messageId": "wrongQuotes",
    "nodeType": "Literal",
    "ruleId": "quotes",
    "severity": 2,
  },
  {
    "column": 5,
    "endColumn": 10,
    "endLine": 19,
    "fix": {
      "range": [
        432,
        467,
      ],
      "text": "color: "blonde",
    length: "long"",
    },
    "line": 19,
    "message": "Expected object keys to be in natural ascending order. 'color' should be before 'length'.",
    "nodeType": "Property",
    "ruleId": "sort-keys-fix/sort-keys-fix",
    "severity": 2,
  },
  {
    "column": 12,
    "endColumn": 20,
    "endLine": 19,
    "fix": {
      "range": [
        459,
        467,
      ],
      "text": "'blonde'",
    },
    "line": 19,
    "message": "Strings must use singlequote.",
    "messageId": "wrongQuotes",
    "nodeType": "Literal",
    "ruleId": "quotes",
    "severity": 2,
  },
  {
    "column": 20,
    "endColumn": 21,
    "endLine": 19,
    "fix": {
      "range": [
        459,
        470,
      ],
      "text": ""blonde" }",
    },
    "line": 19,
    "message": "Unexpected trailing comma.",
    "messageId": "unexpected",
    "nodeType": "Property",
    "ruleId": "comma-dangle",
    "severity": 2,
  },
  {
    "column": 3,
    "endColumn": 6,
    "endLine": 20,
    "fix": {
      "range": [
        424,
        499,
      ],
      "text": "// prettier-ignore
age: 4, 
  hair: { length: "long",
    color: "blonde", }",
    },
    "line": 20,
    "message": "Expected object keys to be in natural ascending order. 'age' should be before 'hair'.",
    "nodeType": "Property",
    "ruleId": "sort-keys-fix/sort-keys-fix",
    "severity": 2,
  },
  {
    "column": 3,
    "endColumn": 4,
    "endLine": 21,
    "fix": {
      "range": [
        502,
        512,
      ],
      "text": ")

export",
    },
    "line": 21,
    "message": "Extra semicolon.",
    "messageId": "extraSemi",
    "nodeType": "VariableDeclaration",
    "ruleId": "@typescript-eslint/semi",
    "severity": 2,
  },
  {
    "column": 23,
    "endColumn": 24,
    "endLine": 23,
    "fix": {
      "range": [
        527,
        529,
      ],
      "text": "]",
    },
    "line": 23,
    "message": "Extra semicolon.",
    "messageId": "extraSemi",
    "nodeType": "ExportDefaultDeclaration",
    "ruleId": "@typescript-eslint/semi",
    "severity": 2,
  },
]
`;

exports[`Base config That the base 'valid' file passes 1`] = `[]`;
